File: src/main/java/io/github/dqualizer/dqlang/types/adapter/request/Request.java
package io.github.dqualizer.dqlang.types.adapter.request;

import com.fasterxml.jackson.annotation.JsonProperty;

import io.github.dqualizer.dqlang.types.dam.Payload;
import io.github.dqualizer.dqlang.types.dam.RequestParameter;
import io.github.dqualizer.dqlang.types.dam.PathVariable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Request {

    private String type;
    private String path;
    @JsonProperty("path_variables")
    private List<PathVariable> pathVariables = new ArrayList<>();
    @JsonProperty("query_params")
    private List<Object> queryParams = new ArrayList<>();
    @JsonProperty("request_params")
    private List<RequestParameter> requestParameters = new ArrayList<>();
    private List<Payload> payload = new ArrayList<>();
    private Checks checks;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/k6/request/Request.kt
package io.github.dqualizer.dqlang.types.adapter.k6.request

import com.fasterxml.jackson.annotation.JsonProperty

data class Request(
  val type: String? = null,

  val path: String? = null,

  @JsonProperty("path_variables")
  val pathVariables: Map<String, String>? = null,

  @JsonProperty("query_params")
  val queryParams: Map<String, String>? = null,

  val params: Map<String, String>? = null,

  val payload: Map<String, String>? = null,

  val checks: Checks? = null
)
